### ISSUE TITLE
 When i upgraded java version to openJDK 11, I am getting nullpointerException while loading modelmapper configurations

### ISSUE THREAD

**Issue by shankarwali** [modelmapper/modelmapper#482]
NullPointerException issue is not resolved even after upgrading modelMapper version to 2.3.5

Error log:

at org.modelmapper.internal.Errors.throwConfigurationExceptionIfErrorsExist(Errors.java:241)
    at org.modelmapper.internal.ExplicitMappingBuilder.build(ExplicitMappingBuilder.java:244)
    at org.modelmapper.internal.ExplicitMappingBuilder.build(ExplicitMappingBuilder.java:96)
    at org.modelmapper.internal.TypeMapImpl.addMappings(TypeMapImpl.java:92)
    at org.modelmapper.internal.TypeMapStore.getOrCreate(TypeMapStore.java:124)
    at org.modelmapper.ModelMapper.addMappings(ModelMapper.java:113)
    ...
Caused by: java.lang.NullPointerException
    at org.modelmapper.internal.ExplicitMappingBuilder$ExplicitMappingInterceptor.access$000(ExplicitMappingBuilder.java:304)
    at org.modelmapper.internal.ExplicitMappingBuilder.createAccessorProxies(ExplicitMappingBuilder.java:287)
    at org.modelmapper.internal.ExplicitMappingBuilder.createProxies(ExplicitMappingBuilder.java:277)
    at org.modelmapper.internal.ExplicitMappingBuilder.visitPropertyMap(ExplicitMappingBuilder.java:266)
    at org.modelmapper.PropertyMap.configure(PropertyMap.java:386)
    at jdk.internal.reflect.GeneratedMethodAccessor16.invoke(Unknown Source)
    at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.base/java.lang.reflect.Method.invoke(Method.java:566)
    at org.modelmapper.internal.ExplicitMappingBuilder.build(ExplicitMappingBuilder.java:227)
    ... 6 more

**Comment by varomorf** [comment]
It is also happening to me with a mapping like:

```
TypeMap<Game, GameRestDTO> typeMap = modelMapper.createTypeMap(Game.class, GameRestDTO.class);

typeMap.addMappings(mapper -> {
	mapper.map(Game::getGameId, GameRestDTO::setId);
	mapper.map(Game::getTitle, GameRestDTO::setTitle);
	mapper.map(game -> game.getGameQuestions().size(), GameRestDTO::setQuestions);
});
```
After debuggin a bit I found that on `org.modelmapper.internal.ProxyFactory` the following exception is thrown:

```
java.lang.IllegalArgumentException: org.modelmapper.internal.bytebuddy.renamed.java.util.List$ByteBuddy$heApCRi1 must be defined in the same package as java.util.List/package
	at org.modelmapper.internal.bytebuddy.dynamic.loading.ClassInjector$UsingLookup.injectRaw(ClassInjector.java:1413)
	at org.modelmapper.internal.bytebuddy.dynamic.loading.ClassInjector$AbstractBase.inject(ClassInjector.java:111)
	at org.modelmapper.internal.bytebuddy.dynamic.loading.ClassLoadingStrategy$UsingLookup.load(ClassLoadingStrategy.java:466)
	at org.modelmapper.internal.bytebuddy.dynamic.TypeResolutionStrategy$Passive.initialize(TypeResolutionStrategy.java:100)
	at org.modelmapper.internal.bytebuddy.dynamic.DynamicType$Default$Unloaded.load(DynamicType.java:5623)
	at org.modelmapper.internal.ProxyFactory.proxyFor(ProxyFactory.java:97)
	at org.modelmapper.internal.ProxyFactory.proxyFor(ProxyFactory.java:72)
	at org.modelmapper.internal.PropertyReferenceCollector$SourceInterceptor.invoke(PropertyReferenceCollector.java:83)
	at org.zapiens.entity.Game$ByteBuddy$Labe7Eub.getGameQuestions(Unknown Source)
	at org.zapiens.api.model.mappers.entities.GameMapper.lambda$configMappings$0(GameMapper.java:62)
	at org.modelmapper.internal.ReferenceMapExpressionImpl.map(ReferenceMapExpressionImpl.java:53)
	at org.modelmapper.internal.ConfigurableConditionExpressionImpl.map(ConfigurableConditionExpressionImpl.java:65)
	at org.zapiens.api.model.mappers.entities.GameMapper.lambda$configMappings$1(GameMapper.java:62)
	at org.modelmapper.internal.TypeMapImpl.addMappings(TypeMapImpl.java:266)
	at org.zapiens.api.model.mappers.entities.GameMapper.configMappings(GameMapper.java:58)
```
This makes the proxies for the list to be null, hence the `NullPointerException`.

Also happens with java.util.Date.

**Comment by varomorf** [comment]
Try to fix it on #522
